### PR TITLE
Rename ExtSlice

### PR DESCRIPTION
**Pull request by DragonMinded**
## Summary

This is part one of a plan to deprecate `ExtSlice` and fix the irregularity of `Subscript.slice` which will address #101. In this diff I simply rename `ExtSlice` to `SubscriptElement`, providing a backwards-compatible shim for existing surfaces using or visiting over `ExtSlice`. This is somewhat complicated by the fact that we need to not just allow construction of nodes/matchers using `ExtSlice`, but we also need to be able to call `visit_ExtSlice` and `leave_ExtSlice` (as well as attribute visitors) on old visitors even though the new node is named `SubscriptElement`. The construction/instance check/matching side of things will work with only the `ExtSlice = SubscriptElement` alias since internally we refer to everything as `SubscriptElement`, but for string-based function lookup, we need to get a little more clever and make the default  visit_SubscriptElement` delegate onward to `visit_ExtSlice` so that either form works.

## Test Plan

tox, write a test to exercise using `ExtSlice` in construction, visiting and matchers.

### CODE REVIEW

File: libcst/_parser/conversions/expression.py [hunk 1]
```diff
@@ -694,15 +694,15 @@ def convert_subscriptlist(
     config: ParserConfig, children: typing.Sequence[typing.Any]
 ) -> typing.Any:
     if len(children) > 1:
-        # This is a list of ExtSlice, so construct as such by grouping every
+        # This is a list of SubscriptElement, so construct as such by grouping every
         # subscript with an optional comma and adding to a list.
         extslices = []
```
Comment (carljm) [inline libcst/_parser/conversions/expression.py]: nit: rename variable to `elements`?

Comment (DragonMinded) [reply libcst/_parser/conversions/expression.py]: Good catch!

**Comment by DragonMinded** [comment]
Yeah, renaming is problematic precisely because of the visitor pattern. `@visit`/`@leave` fixes that as you see here, but they're inherently slower and slightly less ergonomic for general cases. But, its possible!

File: libcst/_typed_visitor.py [hunk 2]
```diff
@@ -4247,6 +4229,41 @@ def visit_Subscript_whitespace_after_value(self, node: "Subscript") -> None:
...
+    def leave_ExtSlice_slice(self, node: "ExtSlice") -> None:
+        pass
+
+    @mark_no_op
+    def visit_ExtSlice_comma(self, node: "ExtSlice") -> None:
+        pass
+
+    @mark_no_op
+    def leave_ExtSlice_comma(self, node: "ExtSlice") -> None:
+        pass
```
Comment (jimmylai) [inline libcst/_typed_visitor.py]: Can we call `warnings.warn(..., DeprecationWarning)` in those deprecated visit and leave function to provide warning to users?

Comment (jimmylai) [reply libcst/_typed_visitor.py]: After rethink about it, for showing the warning, we'll need some work on at_visit/at_leave which doesn't seem worth it.

**Pull request merged**